disperser/batcher/txn_manager.go

```diff
@@ -186,19 +192,24 @@ func (t *txnManager) monitorTransaction(ctx context.Context, req *TxnRequest) (*
 			err = t.ethClient.SendTransaction(ctx, newTx)
 			if err != nil {
 				if errors.Is(err, gcore.ErrNonceTooLow) {
-					// try to get the receipt again to see if the transaction has been mined
-					_, receiptErr := t.ethClient.TransactionReceipt(ctx, req.Tx.Hash())
-					if receiptErr == nil {
-						continue
+					// try to get the receipt again to see if any of the transaction attempts have been mined
+					for _, tx := range req.txAttempts {
+						_, receiptErr := t.ethClient.TransactionReceipt(ctx, tx.Hash())
+						if receiptErr == nil {
+							// tx has been mined, update the request with the mined tx and continue
+							req.Tx = tx
+							continue
+						}
 					}
 				}
 				t.logger.Error("[TxnManager] failed to send txn", "tag", req.Tag, "txn", req.Tx.Hash().Hex(), "err", err)
 				t.metrics.IncrementTxnCount("failure")
 				return nil, err
```

Comment:
If the newTx is failed to sent, this is erroring out directly. Would any of the existing pending transaction still be able to succeed in such case?

Reply:
Any of the existing pending transactions may still succeed even in this case. 
This was meant to catch non-transient errors, so that it doesn't get blocked on a transaction with potential issues. Maybe we should have a retry count here and return error only if it fails n times. 
Thoughts?

Reply:
Yeah that sounds reasonable.
Instead of conditioning on if we can send a new txn, I wonder if it makes sense to:
- periodically  (every txnRefreshInterval) send a new txn (unless a txn is mined)
- whether it succeeds or not, in each cycle we examine all the pending txns for a mined txn
- if this is getting N pending txns, or we have looped M txnRefreshInterval cycles (M may be different than N, because sending a new txn may fail like this), then we return error
- if we find a mined txn, then all other txns cannot succeed since they have same nounce, then we just wait for this mined txn to get enough confirmation, and then return success

I may have missed something, but it seems to be more robust and hopefully the logic in this monitor function is a bit clear?

Reply:
Yes

> whether it succeeds or not, in each cycle we examine all the pending txns for a mined txn

This makes sense

> if this is getting N pending txns, or we have looped M txnRefreshInterval cycles (M may be different than N, because sending a new txn may fail like this), then we return error

Note that when we send a new txn here, it's not a new txn but it's replacing the existing txn. Therefore, even if we replace the txn N times, _once all validators are in sync_, there would be only 1 pending txn. I think treating successful replacement txn vs. failed txn separately still makes sense. If we can successfully increase gas for a pending txn N times (or looped M txnRefreshInterval cycles), that should be allowed to continue being retried. However, if it fails to replace the txn more than N times, I think it's safe to deem it failed. Does it make sense?

Reply:
Ok, sounds good!